services/mgmt/compute/examples/vm_list_using_resource_graph.rs

```diff
@@ -0,0 +1,85 @@
+use std::sync::Arc;
+use azure_identity::DefaultAzureCredential;
+use azure_mgmt_resourcegraph::models::{QueryRequest, QueryRequestOptions};
+use serde_json::{json, Value};
+use azure_mgmt_resourcegraph::Client as AzureResourceGraphClient;
+use azure_mgmt_resourcegraph::ClientBuilder as AzureResourceGraphClientBuilder;
+
+/// Fetching Virtual Machines using Azure Resource Graph
+
+#[derive(Debug)]
+pub enum GenericError {
+    InvalidRequest,
+}
+
+#[derive(Debug)]
+struct CustomError {
+    err_type: GenericError,
+    err_msg: String
+}
+
+async fn fetch_azure_records() -> Result<Value, CustomError> {
+
+    let azure_creds = Arc::new(DefaultAzureCredential::default());
+
+    let az_rg_client = AzureResourceGraphClient::builder(azure_creds).build();
+
+    // fetch first 10 records
+    let query_options = QueryRequestOptions {
+        top: Option::from(10),
+        skip: Option::from(0),
+        ..Default::default()
+    };
+
+    // in the below request, we are fetching resources of type (virtual machines)
+    let custom_query_request = QueryRequest {
+        subscriptions: vec![],
+        management_groups: vec![],
+        query: "Resources | where type == 'microsoft.compute/virtualmachines'".to_string(),
+        options: Option::from(query_options),
+        facets: vec![],
+    };
+
+    let query_request = QueryRequest::from(custom_query_request);
```

Comment:
I don't believe this is needed, as `custom_query_request` is already a `QueryRequest` here.

Reply:
Yep, you can make that change.